### PR TITLE
PTX-2858 Implement ability to deploy/validate PX version based on Spec Generator URL

### PR DESCRIPTION
**Pull request by nikolaypopov**
Implement ability to deploy/validate PX version based on Spec Generator URL which is passed from the command line.

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: test/integration_test/common_test.go [hunk 1]
```diff
@@ -3,27 +3,75 @@
...
+	defaultPxNamespace              = "kube-system"
+	pxReleaseManifestURLEnvVariable = "PX_RELEASE_MANIFEST_URL"
+
+	defaultValidateStorageClusterTimeout       = 900 * time.Second
+	defaultValidateStorageClusterRetryInterval = 30 * time.Second
+	defaultValidateUninstallTimeout            = 900 * time.Second
+	defaultValidateUninstallRetryInterval      = 30 * time.Second
 )
 
+var pxSpecGenURL string
```
Comment (piyush-nimbalkar) [inline test/integration_test/common_test.go:33]: nit: use 
```
var (
  pxSpecGenURL string
  pxEndpoint string
)
```

File: test/integration_test/common_test.go [hunk 2]
```diff
@@ -3,27 +3,75 @@
...
+	flag.StringVar(&pxEndpoint,
+		"portworx-endpoint",
+		"",
+		"Portworx Endpoint that defines the version of Portworx and its components")
+	flag.Parse()
 	os.Exit(m.Run())
 }
 
+// Here we make StorageCluster object and add all the common basic parameters that all StorageCluster should have
+func constructStorageCluster() (*corev1.StorageCluster, error) {
```
Comment (piyush-nimbalkar) [inline test/integration_test/common_test.go:50]: If we want to test a released version of Portworx, how would this work? 
I assume, we would only pass the endpoint in that case. But here we are not setting the portworx version, so the operator will end up installing the latest available version every time, right?

File: test/integration_test/common_test.go (hunk 1 above)
Comment (nikolaypopov) [reply test/integration_test/common_test.go]: Done

File: test/integration_test/common_test.go (hunk 2 above)
Comment (nikolaypopov) [reply test/integration_test/common_test.go:66]: Added setting imageName for all deployments as well as EnvVar in case of `edge`.

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**